qa/lib/schema.ts

```diff
@@ -10,19 +10,41 @@ declare global {
   }
 }
 
```

Comment:
Do/while -- I kind of forgot about that construct! Fancy! Maybe since it's a bit imperative (and we're not testing), I would add this comment for a future self so they don't have to follow the logic:
```suggestion
// Example:  ['hello', 'foo.bar.baz'] -> ['hello', 'foo', 'foo.bar', 'foo.bar.baz']
```